core/src/sumeragi/message.rs

```diff
@@ -143,15 +139,49 @@ impl From<CommittedBlock> for BlockCommitted {
 #[non_exhaustive]
 pub struct BlockSyncUpdate {
     /// The corresponding block.
-    pub block: VersionedCandidateCommittedBlock,
+    block: VersionedCommittedBlock,
 }
 
 impl From<VersionedCommittedBlock> for BlockSyncUpdate {
     fn from(block: VersionedCommittedBlock) -> Self {
-        Self {
-            block: block.into(),
+        Self { block }
+    }
+}
+
+impl BlockSyncUpdate {
+    /// Extract block from block sync update message.
+    ///
+    /// # Errors
+    /// - When the block is invalid.
+    pub fn validate_and_extract_block(
```

Comment:
In my opinion it will be better to enforce type restrictions as early as possible (preferable at system boundary ) and not on the spot.
In our case it will be blocks received as part of `ShareBlocks` message.

Reply:
I don't think that works because block sync does not know if the block is valid or not. Only sumeragi knows because validity is relative to the current state.

Reply:
> Only sumeragi knows because validity is relative to the current state.

Yeah and thus it won't be possible to extract `CommittedBlock` inside block sync so we prevent wrong usage of it.

Reply:
So this conversation can be resolved?